app/screens/auth/CreateWallet.js

```diff
@@ -192,14 +195,15 @@ class CreateWallet extends Component<Props, State> {
     return !passwordError && !verifyPasswordError;
   };
 
-  nextAction = () => {
-    const { history, location } = this.props;
+  nextAction = async () => {
+    const { history, location, getMiningStatus } = this.props;
     const { subMode } = this.state;
     const isWalletOnlySetup = !!location?.state?.withoutNode;
+    const miningStatus = await getMiningStatus();
     if (subMode === 1 && this.validate()) {
       this.createWallet();
     } else if (subMode === 2) {
-      if (isWalletOnlySetup) {
+      if (isWalletOnlySetup || miningStatus !== nodeConsts.NOT_MINING) {
```

Comment:
why is this here? Should it be merged to develop?

Reply:
We want to prevent a case in which we're mining or in the mining setup phase and also redirecting to the nodeSetup section. You suggested last time that we should not have that in the NodeSetup component, but where the redirection occurs, which is here.

Reply:
Ok, then why do you make request for status in click handler? Can it be done before and wrapped by connection check to prevent unnecessary errors and changes in actions?

Reply:
What do you suggest? Can be done in componentDidMount for example. In this instance it's like wallet only setup.

Reply:
Remove getMiningStatus from componentDidMount func in Main and move to healthCheckFlow in App. And rename healthCheckFlow